### DISCUSSION TITLE
Fill Bucket Improvement!

### DISCUSSION THREAD

**Discussion by DinoAwakening** [keiwando/toonsquid#147] [Ideas]
Most of the time, fill bucket won’t get the work done. Especially with those brushes that are so irregular even while anti-aliasing is disabled. What if the bucket tool (And Magic Wand selection tool I mentioned earlier on another discussion), got new options to set margins for filling up areas. Further than only Threshold and Smoothing, Extra Pixels margin (from 1 to 20px), also blur margin to fade the filled area (Making a nice kind of unintentional “Auto-Shade” for edges). Hope Keiwando see this :DD

**Comment by DinoAwakening** [comment]
![84A89337-1027-48C2-A990-9A23F7E149F0](https://user-images.githubusercontent.com/120270913/216848524-7ae753f2-6313-43c2-b539-bd8819e1d00a.jpeg)

RoughAnimator does a magnificent work on filling any shape draw. Also, the extra feature is be able to fill (In front, Behind) in the same layer! :DD

**Reply by jsllllc** [reply]
Coming from RA, I do miss the "fill behind" feature.
RA's "Fill inside" can be replicated in TS by masking, but "fill behind" is most helpful to me as my low-RAM 2017 Ipad Pro can only handle so many layers.

**Reply by DinoAwakening** [reply]
> Coming from RA, I do miss the "fill behind" feature.
> RA's "Fill inside" can be replicated in TS by masking, but "fill behind" is most helpful to me as my low-RAM 2017 Ipad Pro can only handle so many layers.

Keiwan added some sort of “Fill Behind” on 1.8, and “Fill Inside” is just turning on “Alpha Layer”, not even masking is needed. Nevertheless, the important thing here is, the way RoughAnimator allows any kind of illustrator to fill up any kind of area with such an ease. That’s incredible and so unique. I could tell I’ve never seen a bucket tool be so useful in any other app. :D

**Reply by jsllllc** [reply]
I am unclear on how 1.8 allows "fill behind" - I only see changes to fill reference layers. Either way I totaly agree on the "expand fill" and "fill blur" options, both of which I was able to use to great effect in RA.

**Comment by DaleCoooper** [comment]
I totally agree. Fill Expand is absolutely necessary. Closing the thin silhouette gap that always appears after filling is annoying.
